airflow/contrib/hooks/spark_submit_hook.py

```diff
@@ -434,10 +444,13 @@ def _start_driver_status_tracking(self):
             returncode = status_process.wait()
 
             if returncode:
-                raise AirflowException(
-                    "Failed to poll for the driver status: returncode = {}"
-                    .format(returncode)
-                )
+                if retries < max_retries:
```

Comment:
Why do we want to implement the retries inside of the operator. The operator itself also can be retried as a whole.

Reply:
Its a tricky issue.
When you do a Spark Submit to the standalone cluster, you need to constantly poll for the status. Of course, this polling can timeout due to heavy computations on the cluster. Therefore your Airflow job looks like it failed but actually the polling process failed once.
What I found is that a heavy computation job will constantly fail due to this timeout and therefore never succeed although it is completed on the cluster.

Reply:
Isn;t there another way of checking for a succesful execution then? Maybe something that you expect at the end of the job to be present? I am not a fan of how it is implemented now.

Reply:
This issue is not job specific.
To check if a SparkJob has been successful on the cluster, you need to poll its status. Because the polling is a network request, it makes sense to have a retry mechanism.
In a distributed context you cannot expect your network requests to a heavily loaded cluster to always be successful on the first try.

Reply:
If I rename the variables `max_retries` to `max_missed_job_status_reports`, does that provide more clarity?

Reply:
@Fokko @bolkedebruin

Reply:
Yes, sounds like a good idea. Change it to `max_missed_job_status_reports` and add a comment about the issue (timeout at polling). Personally I would say that your machine/cluster isn't configured correctly, you might want to reserve more resources for these kind of calls. This would also mean that other service in the network/cluster would suffer tremendously from the running Spark jobs, (not to mention stuff like HDFS).